package/src/types.ts

```diff
@@ -56,11 +56,12 @@ export type GuardFunction = (
   from: GuardFunctionRouteProps | null,
   next: Next,
 ) => void;
+export type RouteError = string | null;
 
 /**
  * Page Component Types
  */
-export type PageComponent = ComponentType | null | undefined | string | boolean | number;
+export type PageComponent<P = {}> = ComponentType<P> | null | undefined | string | boolean | number;
```

Comment:
Since these pages also get the route props, we should include those here too

```suggestion
export type PageComponent<P = {}> = ComponentType<RouteProps & P> | null | undefined | string | boolean | number;
```